### PR TITLE
HPCC-21343 Check File Permissions screen shows error when opened

### PR DESCRIPTION
**Pull request by kunalaswani**
Required value field was not clearing in dialog, form resets to clear dialog required fields. Fixes HPCC-21344 (Choosing one file permission should clear the opposing dialog) .

Signed-off-by: Kunal Aswani <kunal.aswani@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by kunalaswani** [comment]
@miguelvazq please review

File: esp/src/eclwatch/UserQueryWidget.js [hunk 1]
```diff
@@ -157,6 +157,7 @@ define([
         _onCheckFilePermissions: function () {
             var context = this;
             this.filePermissionDialog.show();
+            this.filePermissionsForm.reset();
```
Comment (miguelvazq) [inline esp/src/eclwatch/UserQueryWidget.js]: This be done onClose.

**Review by miguelvazq — CHANGES_REQUESTED** [review]

File: esp/src/eclwatch/UserQueryWidget.js (hunk 1 above)
Comment (kunalaswani) [reply esp/src/eclwatch/UserQueryWidget.js]: i submitted it backwards, but when set on close, it only resets when the user clicks the close button, however if the click "x" on the dialog box that function doesnt clear the dialog. this ensure that the form is always cleared.

Comment (miguelvazq) [reply esp/src/eclwatch/UserQueryWidget.js]: Yes, but the data appears and then just disappears on open. Here is the API for the dijit/Dialog for all cancel,hide,close options. https://dojotoolkit.org/api/#1_10dijit_Dialog_onCancel

**Comment by kunalaswani** [comment]
Please review @miguelvazq changes made to dialog

**Comment by miguelvazq** [comment]
Looks good now.

**Comment by miguelvazq** [comment]
Looks good

**Pull request merged**